### PR TITLE
Setup empty api package

### PR DESCRIPTION
**Pull request by RainbowBunchie**
depends on #2 

Setup the basic (empty) api package

Closes COM-265

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ schema.graphql
 comet-lang
 /api/uploads
 lang
+lib
```
Comment (johnnyomair) [inline .gitignore:12]: Please move this to the respective .gitignore of the package

File: packages/api/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "@comet/eslint-config/nestjs",
+    "ignorePatterns": ["src/mikro-orm/migrations/**", "lib/**"],
```
Comment (johnnyomair) [inline packages/api/.eslintrc.json:3]:
```suggestion
    "ignorePatterns": ["lib/**"],
```

We don't ignore migrations in the Starter anymore, so we shouldn't here either

File: packages/api/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    "rules": {
+        "@comet/no-other-module-relative-import": "off"
```
Comment (johnnyomair) [inline packages/api/.eslintrc.json:5]: Why do you need to deactivate this rule?

File: packages/api/bin/comet-api-generator.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env node
```
Comment (johnnyomair) [inline packages/api/bin/comet-api-generator.js:1]: We don't need this here

File: packages/api/list-exports.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import * as all from "./src";
```
Comment (johnnyomair) [inline packages/api/list-exports.ts:1]: Please also delete this

File: packages/api/package.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+    "name": "@comet-brevo-module/api",
```
Comment (johnnyomair) [inline packages/api/package.json:2]: This should be `@comet/brevo-module-api` I suppose, or even `@comet/brevo-api`?

File: packages/api/package.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        "type": "git",
+        "url": "https://github.com/vivid-planet/comet-brevo-module/",
+        "directory": "packages/api/"
+    },
+    "license": "BSD-2-Clause",
+    "main": "lib/index.js",
+    "types": "lib/index.d.ts",
+    "bin": {
+        "comet-api-generator": "bin/comet-api-generator.js"
+    },
```
Comment (johnnyomair) [inline packages/api/package.json:14]:
```suggestion
```

File: packages/api/package.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    "files": [
+        "bin/**/*.js",
```
Comment (johnnyomair) [inline packages/api/package.json:16]:
```suggestion
```

File: packages/api/package.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        "dev": "tsc --watch --preserveWatchOutput -p tsconfig.build.json",
+        "lint": "run-p lint:eslint lint:tsc",
+        "lint:eslint": "eslint --max-warnings 0 src/ package.json",
+        "lint:tsc": "tsc --noEmit",
+        "list-exports": "ts-node list-exports.ts",
+        "test": "jest",
+        "test:watch": "jest --watch"
+    },
+    "dependencies": {
+        "rimraf": "^3.0.0"
```
Comment (johnnyomair) [inline packages/api/package.json:31]: This should be a dev dependency

File: packages/api/package.json [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "devDependencies": {
+        "@comet/eslint-config": "^5.3.0",
+        "@nestjs/common": "^9.0.0",
```
Comment (johnnyomair) [inline packages/api/package.json:35]: You'll probably need some peerDependencies here

File: packages/api/src/comet-brevo-module.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { DynamicModule, Global, Module } from "@nestjs/common";
+
+@Global()
+@Module({})
+export class CometBrevoModule {
```
Comment (johnnyomair) [inline packages/api/src/comet-brevo-module.ts:5]: I'd drop the Comet prefix, just call it BrevoModule

File: .gitignore (hunk 1 above)
Comment (RainbowBunchie) [reply .gitignore]: moved in 8dfd38991f975e0390502a413279292dc8fdb117

File: packages/api/.eslintrc.json (hunk 2 above)
Comment (RainbowBunchie) [reply packages/api/.eslintrc.json]: I removed this in 342f4bf10c40e957ca89d89de2e1877ed50a6938

File: packages/api/.eslintrc.json (hunk 3 above)
Comment (RainbowBunchie) [reply packages/api/.eslintrc.json]: This is not needed here. I removed this in 342f4bf10c40e957ca89d89de2e1877ed50a6938

File: packages/api/bin/comet-api-generator.js (hunk 4 above)
Comment (RainbowBunchie) [reply packages/api/bin/comet-api-generator.js]: I removed this in c94ff3cacc315c168cf437a2f8d6b45a0f7c96b6

File: packages/api/list-exports.ts (hunk 5 above)
Comment (RainbowBunchie) [reply packages/api/list-exports.ts]: I removed this in c94ff3cacc315c168cf437a2f8d6b45a0f7c96b6

File: packages/api/package.json (hunk 6 above)
Comment (RainbowBunchie) [reply packages/api/package.json]: I changed the name to `@comet/brevo-api` in 38a792aff39ac7019ec5c9155c85e173e3419c8b

File: packages/api/package.json (hunk 7 above)
Comment (RainbowBunchie) [reply packages/api/package.json]: I removed this in c94ff3cacc315c168cf437a2f8d6b45a0f7c96b6

File: packages/api/package.json (hunk 8 above)
Comment (RainbowBunchie) [reply packages/api/package.json]: I removed this in c94ff3cacc315c168cf437a2f8d6b45a0f7c96b6

File: packages/api/package.json (hunk 9 above)
Comment (RainbowBunchie) [reply packages/api/package.json]: I moved this to dev dependencies in [44f62c5](https://github.com/vivid-planet/comet-brevo-module/pull/3/commits/44f62c58dded7a8135cbc3d6b3a71e8e7ee918a2)

File: packages/api/package.json (hunk 10 above)
Comment (RainbowBunchie) [reply packages/api/package.json:27]: As of now this works without further peerDependencies

File: packages/api/src/comet-brevo-module.ts (hunk 11 above)
Comment (RainbowBunchie) [reply packages/api/src/comet-brevo-module.ts]: I dropped the prefix in [38a792a](https://github.com/vivid-planet/comet-brevo-module/pull/3/commits/38a792aff39ac7019ec5c9155c85e173e3419c8b)

File: packages/api/package.json (hunk 10 above)
Comment (johnnyomair) [reply packages/api/package.json:27]: `@nestjs/common`, `reflect-metadata` and `rxjs` should be both peer dependencies (for projects) and dev dependencies (for development).

Comment (RainbowBunchie) [reply packages/api/package.json:32]: I added those a peers in  [9a54cbc](https://github.com/vivid-planet/comet-brevo-module/pull/3/commits/9a54cbcfed81492e19850a0cbf2779887f3414f5)

**Pull request merged**